presto-kafka/src/main/java/com/facebook/presto/kafka/KafkaRecordSet.java

```diff
@@ -305,30 +308,26 @@ private void checkFieldType(int field, Class<?> expected)
         @Override
         public void close()
         {
+            if (consumer != null) {
+                consumer.close();
+            }
         }
 
         private void openFetchRequest()
         {
             try {
                 if (messageAndOffsetIterator == null) {
-                    log.debug("Fetching %d bytes from offset %d (%d - %d). %d messages read so far", KAFKA_READ_BUFFER_SIZE, cursorOffset, split.getStart(), split.getEnd(), totalMessages);
-                    FetchRequest req = new FetchRequestBuilder()
-                            .clientId("presto-worker-" + Thread.currentThread().getName())
-                            .addFetch(split.getTopicName(), split.getPartitionId(), cursorOffset, KAFKA_READ_BUFFER_SIZE)
-                            .build();
-
-                    // TODO - this should look at the actual node this is running on and prefer
-                    // that copy if running locally. - look into NodeInfo
-                    SimpleConsumer consumer = consumerManager.getConsumer(split.getLeader());
-
-                    FetchResponse fetchResponse = consumer.fetch(req);
-                    if (fetchResponse.hasError()) {
-                        short errorCode = fetchResponse.errorCode(split.getTopicName(), split.getPartitionId());
-                        log.warn("Fetch response has error: %d", errorCode);
-                        throw new RuntimeException("could not fetch data from Kafka, error code is '" + errorCode + "'");
+                    long threadId = Thread.currentThread().getId();
+
+                    if (consumer == null) {
+                        consumer = consumerManager.createConsumer(threadId, split.getLeader());
```

Comment:
use Thread.currentThread().getName(), instead of creating threadId

Reply:
Fixed.